cartographer_ros/cartographer_ros/node_main.cc

```diff
@@ -38,14 +41,20 @@ void Run() {
   tf2_ros::TransformListener tf(tf_buffer);
   NodeOptions node_options;
   TrajectoryOptions trajectory_options;
-  std::tie(node_options, trajectory_options) =
-      LoadOptions(FLAGS_configuration_directory, FLAGS_configuration_basename);
-
+  if (FLAGS_start_trajectory_with_default_topics) {
+    std::tie(node_options, trajectory_options) = LoadOptions(
+        FLAGS_configuration_directory, FLAGS_configuration_basename);
+  } else {
+    node_options = LoadNodeOptions(FLAGS_configuration_directory,
```

Comment:
Please do not move this into `node_main.cc` and have a look at other uses first. Loading options is also done in the offline node and in the configuration file tests in this and other Cartographer repos.